src/proof-of-concept.java

```diff
@@ -0,0 +1,174 @@
+import components.queue.Queue1L;
+import components.queue.Queue;
+import components.sequence.Sequence;
+import components.sequence.Sequence1L;
+import components.simplewriter.SimpleWriter;
+import components.simplewriter.SimpleWriter1L;
+
+public class UserStorage {
+
+    /**
+     * User class for tracking information.
+     */
+    private final class User {
```

Comment:
You can make this a separate file in your design if you'd like. Right now it's an inner class, but I would be happy with it as a "java record".